crates/runtime/src/vmcontext.rs

```diff
@@ -11,24 +11,57 @@ use std::marker;
 use std::ptr::NonNull;
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::u32;
-pub use vm_host_func_context::VMHostFuncContext;
-use wasmtime_environ::DefinedMemoryIndex;
+pub use vm_host_func_context::{VMArrayCallHostFuncContext, VMNativeCallHostFuncContext};
+use wasmtime_environ::{DefinedMemoryIndex, VMCONTEXT_MAGIC};
 
-pub const VMCONTEXT_MAGIC: u32 = u32::from_le_bytes(*b"core");
+/// A function pointer that exposes the calling convention.
+pub type VMArrayCallFunction =
+    unsafe extern "C" fn(*mut VMOpaqueContext, *mut VMOpaqueContext, *mut ValRaw, usize);
```

Comment:
Could this and the two types below have some more documentation about the ABI? E.g. for the structs below a definition of the ABI and for this function pointer a definition of what each of the arguments are?